internal/cli/atlas/backup/exports/jobs/watch.go

```diff
@@ -0,0 +1,99 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jobs
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+type WatchOpts struct {
+	cli.GlobalOpts
+	cli.WatchOpts
+	id          string
+	clusterName string
+	store       store.ExportJobsDescriber
+}
+
+func (opts *WatchOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+func (opts *WatchOpts) watcher() (bool, error) {
+	result, err := opts.store.ExportJob(opts.ConfigProjectID(), opts.clusterName, opts.id)
+	if err != nil {
+		return false, err
+	}
+	return result.State == "Successful" || result.State == "Failed" || result.State == "Cancelled", nil
+}
+
+func (opts *WatchOpts) Run() error {
+	if err := opts.Watch(opts.watcher); err != nil {
+		return err
+	}
+
+	return opts.Print(nil)
+}
+
+// WatchBuilder atlas backup(s) export(s) job(s) watch <exportJobId>.
+func WatchBuilder() *cobra.Command {
+	opts := new(WatchOpts)
+	cmd := &cobra.Command{
+		Use:   "watch <bucketId>",
+		Short: "Watch for an export job to complete.",
+		Long: `This command checks the export job's status periodically until it reaches a completed, cancelled or failed status. 
+Once the export reaches the expected status, the command prints "Export completed."
+If you run the command in the terminal, it blocks the terminal session until the resource status completes or fails.
+You can interrupt the command's polling at any time with CTRL-C.`,
+		Args: require.ExactArgs(1),
+		Annotations: map[string]string{
+			"bucketIdDesc": "Unique string that identifies the AWS S3 bucket to which you export your snapshots.",
+		},
+		Example: fmt.Sprintf(`  # Watch the continuous backup restore job with the ID 507f1f77bcf86cd799439011 for the cluster named Cluster0 until it becomes available:
+  %s backup restore watch 507f1f77bcf86cd799439011 --clusterName Cluster0`, cli.ExampleAtlasEntryPoint()),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			return opts.PreRunE(
+				opts.ValidateProjectID,
+				opts.initStore(cmd.Context()),
+				opts.InitOutput(cmd.OutOrStdout(), "\nExport completed.\n"),
```

Comment:
do we get `Export completed` regardless of how the export completed? (success, failed, cancelled)
If yes, I think we should display the actual status

Reply:
All watch commands work the same, meaning we don't have different messages based on status, see an example on `atlas clusters watch`

Reply:
I think is a good idea to propose our watch commands return the actual status

Reply:
in the case of `atlas clusters watch` there is only one success condition `return result.StateName == "IDLE", nil`, but for multiple exit conditions (apart from errors), I'd say it's bad user experience to not show the actual completion status.
But you're right, other similar commands do the same

Reply:
https://jira.mongodb.org/browse/CLOUDP-167723